client/state/data-layer/wpcom/discussions/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import debug from 'debug';
+import { mergeHandlers } from 'state/data-layer/utils';
+import {
+	DISCUSSIONS_REQUEST,
+	DISCUSSIONS_ITEM_LIKE_REQUEST,
+	DISCUSSIONS_ITEM_UNLIKE_REQUEST,
+	DISCUSSIONS_ITEM_STATUS_UPDATE_REQUEST,
+	DISCUSSIONS_ITEM_CONTENT_UPDATE_REQUEST
+} from 'state/action-types';
+import {
+	requestingPostComments,
+	receivePostComments,
+	successPostCommentsRequest,
+	failPostCommentsRequest,
+	receivePostCommentsCount,
+} from 'state/discussions/actions';
+import {
+	requestCommentLike,
+	requestCommentUnLike
+} from './likes';
+import requestCommentStatusUpdate from './status';
+import content from './content';
+
+/**
+ * Module variables
+ */
+const log = debug( 'calypso:middleware-comments' );
+
+ /**
+  * Dispatches a request to fetch all available discussions for a post with query options
+  *
+	* @param   {Function} dispatch Redux dispatcher
+  * @param   {Object}   action   Redux action
+  * @param   {Function} next     data-layer-bypassing dispatcher
+  * @returns {Promise}           Promise
+  */
+export const requestDiscussions = ( { dispatch }, action ) => {
+	const {
+		siteId,
+		postId,
+		status
+	} = action;
+
+	log( 'Request discussions for post %d on site %d using status %o', postId, siteId, status );
+
+	dispatch( requestingPostComments( siteId, postId, status ) );
+
+	return wpcom.site( siteId )
```

Comment:
Looks like these actions aren't using the http layer. Would it be easier if we scoped down the initial PR to the actions you already converted? It'll be easier to test and faster for this to land.